pulsar-common/src/main/java/com/yahoo/pulsar/common/api/Commands.java

```diff
@@ -429,11 +456,36 @@ private static ByteBuf serializeCommandSendWithSize(BaseCommand.Builder cmdBuild
             cmd.writeTo(outStream);
             cmd.recycle();
             cmdBuilder.recycle();
+            
+            //Write checksum
+            if (includeChecksum) {
+                headers.writeShort(magicCrc32c);
+                checksumReaderIndex = headers.writerIndex();
+                headers.writeZero(4); // write dummy checksum int to skip 4 bytes in write index
+            }
 
+            int metadataReaderIndex = headers.writerIndex();
             // Write metadata
             headers.writeInt(msgMetadataSize);
             msgMetadata.writeTo(outStream);
 
+            if (includeChecksum) {
+                // mark reader and write index
+                int headersReaderIndex = headers.readerIndex();
+
+                // compute checksum
+                headers.retain();
+                payload.retain();
+                headers.readerIndex(metadataReaderIndex);
+                ByteBuf msgMetadataBuf = DoubleByteBuf.get(headers, payload);
```

Comment:
Actually there are two things:
- [Circe java](https://github.com/trevorr/circe/blob/master/circe-impl/src/main/java/com/scurrilous/circe/impl/AbstractIncrementalIntHash.java#L91) implementation doesn't have incremental checksum support. 
```
 public int resume(int current, long address, long length) {
        throw new UnsupportedOperationException();
  }
```

- and client side: payload is type of ```UnpooledHeapByteBuf```. So, we can't use ```memoryAddress``` so, java implementation also doesn't have incremental checksum using array.